onebusaway-transit-data-federation/src/main/java/org/onebusaway/transit_data_federation/impl/beans/RouteScheduleBeanServiceImpl.java

```diff
@@ -158,8 +160,8 @@ private void addSituations(RouteScheduleBean rsb, AgencyAndId routeId) {
   private void addStopTripDirectionsViaBlockTrip(RouteScheduleBean rsb, AgencyAndId routeId) {
     List<BlockTripIndex> blockTripIndices = _blockIndexService.getBlockTripIndicesForRouteCollectionId(routeId);
 
-    Map<DirectionalHeadsign, StopTripDirectionBean> headsignToBeanMap = new HashMap<>();
-    Map<DirectionalHeadsign, StopCollections> headsignToStopCollectionMap = new HashMap<>();
+    Map<Double, StopTripDirectionBean> headsignToBeanMap = new HashMap<>();
+    Map<Double,DirectionHeadsignStopCollection> directionHeadsignStopCollectionHashMap = new HashMap();
```

Comment:
I understand what you are trying to do here, but Double in Java isn't "stable" and doesn't make a great key for HashMaps.  Read https://stackoverflow.com/questions/1074781/double-in-hashmap  for the backstory.

If you implement equals/hashcode on the DirectionHeadsignStopCollection, you can use the object itself as the key.  That may clean up things quite a bit.